pkg/collector/corechecks/snmp/value.go

```diff
@@ -0,0 +1,43 @@
+package snmp
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"strconv"
+)
+
+type snmpValueType struct {
+	submissionType metrics.MetricType // used when sending the metric
+	value          interface{}        // might be a `string` or `float64` type
```

Comment:
too bad there's not Union type in golang as in TS (or at least I could not google it)

Reply:
unfortunately not possible for now in go :(